src/components/views/HomePage.jsx

```diff
@@ -1,12 +1,15 @@
 import React, { PureComponent } from 'react';
-import HeroSection from '../hero-section/HeroSection';
+import HeroSection from '../containers/hero-section/HeroSection';
 import Navbar from '../containers/navbar/navbar';
 import RecentArticles from '../containers/recentArticles/RecentArticles';
 import FeaturedAuthor from '../containers/featuredAuthor/FeaturedAuthor';
 import TrendingSection from '../trendingsection/TrendingSection';
+import getAndSetUrlToken from '../../utilities/urlTokenHandler';
 
 class HomePage extends PureComponent {
   render() {
+    const { href } = window.location;
+    getAndSetUrlToken(href);
```

Comment:
Can't you do this in componentDidmount?

Reply:
We can.

Reply:
so please do it there

Reply:
I have now. Just realized that I have not pushed it.